dns.c

```diff
@@ -47,18 +51,38 @@ int type;
  errno = 0;
  if (!stralloc_copy(&glue,domain)) return DNS_MEM;
  if (!stralloc_0(&glue)) return DNS_MEM;
- responselen = lookup(glue.s,C_IN,type,response.buf,sizeof(response));
+ if (!responsebuflen) {
+  if ((response.buf = (unsigned char *)alloc(PACKETSZ+1)))
```

Comment:
replace that with malloc() as this is "new" code. The cast isn't needed then anymore as malloc() returns void*

Reply:
Oops, I was sure I'd gotten that. Good catch. I've gone a little further and gotten rid of `alloc.h`.